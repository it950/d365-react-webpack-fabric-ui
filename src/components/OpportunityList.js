import React from 'react'

import OpportunityListItem from './OpportunityListItem'

const OpportunityList = ({ opportunities }) => (
  <ul>
    {opportunities.map(o => (
      <OpportunityListItem key={o.opportunityid} opportunity={o} />
    ))}
  </ul>
)

export default OpportunityList
