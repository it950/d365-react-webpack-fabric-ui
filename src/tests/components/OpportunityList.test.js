import React from 'react'
import { shallow } from 'enzyme'

import OpportunityList from '../../components/OpportunityList'

describe('OpportunityList', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<OpportunityList />)
    expect(wrapper).toMatchSnapshot()
  })
})
