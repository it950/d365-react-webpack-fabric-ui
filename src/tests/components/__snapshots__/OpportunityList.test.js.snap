// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpportunityList should render correctly with no opportunities 1`] = `<ul />`;

exports[`OpportunityList should render correctly with opportunities 1`] = `
<ul>
  <OpportunityListItem
    key="1001"
    opportunity={
      Object {
        "createdon": "2017-9-21 00:00:00",
        "name": "Big Sale",
        "opportunityid": 1001,
      }
    }
  />
  <OpportunityListItem
    key="1002"
    opportunity={
      Object {
        "createdon": "2017-10-20 00:00:00",
        "name": "2018 Restocking",
        "opportunityid": 1002,
      }
    }
  />
  <OpportunityListItem
    key="1003"
    opportunity={
      Object {
        "createdon": "2017-6-15 00:00:00",
        "name": "2017 Q4 Reorder",
        "opportunityid": 1003,
      }
    }
  />
  <OpportunityListItem
    key="1004"
    opportunity={
      Object {
        "createdon": "2017-12-1 00:00:00",
        "name": "End-of-Year Firesale",
        "opportunityid": 1004,
      }
    }
  />
</ul>
`;
